pandas/core/internals/blocks.py

```diff
@@ -1887,7 +1887,7 @@ def take_nd(self, indexer, axis=0, new_mgr_locs=None, fill_tuple=None):
                                       allow_fill=True)
 
         # if we are a 1-dim object, then always place at 0
-        if self.ndim == 1:
+        if self.ndim == 1 and new_mgr_locs is None:
```

Comment:
I have no idea if this is going to break things.

Oh and I notice that the comment is out of date now.